src/b64.mli

```diff
@@ -24,23 +26,50 @@
 
     {e Release %%VERSION%% - %%PKG_HOMEPAGE%%} *)
 
-val default_alphabet : string
-(** A 64-character string specifying the regular Base64 alphabet. *)
+type alphabet
+type sub = string * int * int
 
-val uri_safe_alphabet : string
-(** A 64-character string specifying the URI- and filename-safe Base64
+val default_alphabet : alphabet
+(** A 64-character alphabet specifying the regular Base64 alphabet. *)
+
+val uri_safe_alphabet : alphabet
+(** A 64-character alphabet specifying the URI- and filename-safe Base64
     alphabet. *)
 
-val decode : ?alphabet:string -> string -> string
+val make_alphabet : string -> alphabet
+(** Make a new alphabet. *)
+
+val length_alphabet : alphabet -> int
+(** Returns length of the alphabet, should be 64. *)
+
+val alphabet : alphabet -> int array
+(** Returns the alphabet. *)
+
+val decode_exn : ?pad:bool -> ?alphabet:alphabet -> ?off:int -> ?len:int -> string -> string
 (** [decode s] decodes the string [s] that is encoded in Base64 format. Will
```

Comment:
minor: should be `[decode_exn s]`. Might as well mention the purpose of `off` and `len` too.